BraintreeCore/src/main/java/com/braintreepayments/api/ShopperInsightsRequest.kt

```diff
@@ -0,0 +1,65 @@
+package com.braintreepayments.api
+
+import org.json.JSONObject
+
+/**
+ * Representation of a user phone number.
+ * @property countryCode The international country code for the shopper's phone number
+ * (e.g., "1" for the United States).
+ * @property nationalNumber The national segment of the shopper's phone number
+ * (excluding the country code).
+ */
+
+data class BuyerPhone(
+    var countryCode: String,
+    var nationalNumber: String
+) {
+    companion object {
+        internal const val KEY_COUNTRY_CODE = "countryCode"
+        internal const val KEY_NATIONAL_NUMBER = "nationalNumber"
+    }
+
+    fun toJson(): JSONObject = JSONObject().apply {
+            put(KEY_COUNTRY_CODE, countryCode)
+            put(KEY_NATIONAL_NUMBER, nationalNumber)
+        }
+}
+
+/**
+ * Data class representing a request for shopper insights.
+ *
+ * Note: **This feature is in beta. It's public API may change in future releases.**
+ */
+sealed class ShopperInsightsRequest {
+    data class Email(var email: String) : ShopperInsightsRequest()
+    data class Phone(
+        var phone: BuyerPhone
+    ) : ShopperInsightsRequest()
+    data class EmailAndPhone(
+        var email: String,
+        var phone: BuyerPhone
+    ) : ShopperInsightsRequest()
```

Comment:
I'm not sure I understand the purpose of these data classes? Can we just have a ShopperInsightsRequest class with `email: String?`, `phone: BuyerPhone?` properties?

Reply:
+1 

It could be a little confusing to merchants on which class is needed based on what info they want to pass in. I think the nullable fields route may work. We'd just need to do a check to see if both fields are null and handle that case.

Reply:
Primarily from a merchant integration standpoint, it's not a pattern we use anywhere else in the SDK for request objects - we typically have configurable request objects with multiple required or optional parameters that can be set via request properties. This pattern also seems pretty limiting for any future configurable parameters (ex: if we add a user name property, then do we need to add every possible combination of email/phone/name as data classes?)

Reply:
> We'd just need to do a check to see if both fields are null and handle that case.
Handling that case would be throw an exception to the merchant who has to look out for it and add more code to handle it.

IMHO this approach is cleaner.

Reply:
> This pattern also seems pretty limiting for any future configurable parameters

Agreed.

Reply:
I agree going the sealed class route is cleaner with not having to handle the null checks, but still feel the interface is not as easy to work with than having the optional email and phone fields.

A merchant may think that multiple `ShopperInsightsRequest` can be passed in if they see `Email` and `Phone` fields. 

I think we could use the existing result callback to notify the merchant of the error due to both of the fields being null. They shouldn't need to add additional logic to handle this specific case.

Reply:
changed